app/src/main/java/com/support/android/designlibdemo/activities/SignPetitionActivity.java

```diff
@@ -35,11 +46,47 @@
     TextView tvPetitionMessage;
 
 
+    public String[] getGeolocation() {
+        LocationManager lm = (LocationManager) getSystemService(Context.LOCATION_SERVICE);
+        Location location = lm.getLastKnownLocation(LocationManager.GPS_PROVIDER);
```

Comment:
Emulator crashes on this line, therefore temp fix would be 
double longitude = -122.419;
        double latitude = 37.7749;

        LocationManager lm = (LocationManager) getSystemService(Context.LOCATION_SERVICE);
        Location location = lm.getLastKnownLocation(LocationManager.GPS_PROVIDER);

        if(location!=null){
            longitude = location.getLongitude();
            latitude = location.getLatitude();
        }